translated warnings/dont-call-proptypes (#121)
reactjs/ar.reactjs.org: content/warnings/dont-call-proptypes.md

permalink: warnings/dont-call-proptypes.html
---

> ملاحظة:
>

> > تم ترحيل `React.PropTypes` إلى حزمة منفصلة منذ نسخة React v15.5. الرجاء استخدام مكتبة `prop-types` عوضًا عن ذلك.

Reviewer: Missing link to the library

```suggestion
> تم ترحيل `React.PropTypes` إلى حزمة منفصلة منذ نسخة React v15.5. الرجاء استخدام  [مكتبة `prop-types`](https://www.npmjs.com/package/prop-types) عوضًا عن ذلك.
```

>
> نحن نوفر [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes) لأتمَتَت التحويل.

في إصدار رئيسي مستقبلي من React ، سيتم إزالة الشيفرة التي تطبق دوال التحقق من انواع الخواص في نُسخة الإنتاج. عند حدوث ذلك ، أي شيفرة تستدعي هذه الدوال -التي لم يتم إزالتها في نسخة الإنتاج- يدويًا سيُحدث "يُلقي" خطأ.

### لا بأس بتصريح PropTypes {#declaring-proptypes-is-still-fine}

استخدام PropTypes الإعتيادي ما زال مدعومًا:

```javascript
Button.propTypes = {
  highlighted: PropTypes.bool
};
```

> لا تَغيُر هنا.

Author: yeah, I was kinda sceptical about that one but I didn't know what to do ... perhaps we should take another opinion 😃

> لا تَغيُر هنا.

Author: is this a duplicate @Fcmam5

Reviewer: Sorry, I deleted my comments here

> لا تَغيُر هنا.

Reviewer: Perfect! @iRayan7 can you check this :) 
For me,  I rechecked and I found these example on [Almaany.com](https://www.almaany.com/ar/dict/ar-ar/%D8%AA%D8%BA%D9%8A%D8%B1/):
<div dir="rtl"><code> حَدَثَ تَغَيُّرٌ مُفَاجِئٌ فِي طَقْسِ اليَوْمِ : تَبَدُّلٌ ، تَحَوُّلٌ </code></div>
And even my suggestion of the word `تغييرات` could be wrong, I've updated it to be `تغيّرات`

> لا تَغيُر هنا.

Author: @Fcmam5 maybe it's time we settle ... I agree with your updated choice, I'm gonna commit it

}
```

في هذه الحالة ، `ThirdPartyPropTypes.deprecated` مُغَلِّف "wrapper" يستدعي `PropTypes.bool`. لا بأس بهذا النمط لوحده ولكنه يُطلق إيجابية زائفة "false positive" لأن React تعتقد انك تستدعي PropTypes مُباشرة. المَقطع التالي سيشرح كيفية إصلاح هذه المُشكِلة لِمَكتبة تُطَبِق  شيئًا مِثل `ThirdPartyPropTypes`. إن لَم تَكُن مُؤَلِّف المكتبة يُمكِنُكَ تقديم طلب مُشكلة "issue" ضدها.

> ### إصلاح الأيجابية الزائفة في PropTypes الطرف الثالث {#fixing-the-false-positive-in-third-party-proptypes}

Reviewer: ```suggestion
### إصلاح الإيجابيّة الزائفة في PropTypes الطرف الثالث {#fixing-the-false-positive-in-third-party-proptypes}
```

> إن كُنت مُؤَلِف مكتبة PropTypes طرف ثالث و كنت تسمح للمُستَخدمين بِتَغليف PropTypes من React الموجودة ، فأنهم قد يَرون هذا التحذير آتيًا من المكتبة خاصتك. يَحُدث هذا لأن React لا تَرى الوسيط "argument" الأخير "سريّ secret" [التي تُمَرِره](https://github.com/facebook/react/pull/7132) لِرَصد استدعائات PropTypes اليدوية.  

Reviewer: ```suggestion
إن كُنت مُؤَلِّف مكتبة PropTypes طرف ثالث و كنت تسمح للمُستَخدمين بِتَغليف PropTypes من React الموجودة ، فأنهم قد يَرون هذا التحذير آتيًا من المكتبة خاصتك. يَحُدث هذا لأن React لا تَرى الوسيط "argument" الأخير "سريّ secret" [التي تُمَرِره](https://github.com/facebook/react/pull/7132) لِرَصد استدعائات PropTypes اليدوية.  
```

> إن كُنت مُؤَلِّف مكتبة PropTypes طرف ثالث و كنت تسمح للمُستَخدمين بِتَغليف PropTypes من React الموجودة ، فأنهم قد يَرون هذا التحذير آتيًا من المكتبة خاصتك. يَحُدث هذا لأن React لا تَرى الوسيط "argument" الأخير "سريّ secret" [التي تُمَرِره](https://github.com/facebook/react/pull/7132) لِرَصد استدعائات PropTypes اليدوية.  

Reviewer: Wouldn't be fine just like this? :smiley: 

```suggestion
إن كُنت مُؤَلِّف مكتبة PropTypes طرف ثالث و كنت تسمح للمُستَخدمين بِتَغليف PropTypes من React الموجودة ، فقد يَرون هذا التحذير آتيًا من المكتبة خاصتك. يَحُدث هذا لأن React لا تَرى الوسيط "argument" الأخير "سريّ secret" [التي تُمَرِره](https://github.com/facebook/react/pull/7132) لِرَصد استدعائات PropTypes اليدوية.  
```
